tests/expected/arbitrary/ptrs/pointer_inbounds.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//
+// kani-flags: -Z mem-predicates
+//! Check different cases for `PointerGenerator` for in-bounds pointers.
+//! TODO: Enable initialization checks once we enable reasoning about unions.
```

Comment:
I'm curious: where is "reasoning about unions" disabled?

Reply:
I'm talking about enabling `-Z uninit-checks`. Kani memory initializatin instrumentation does not work in the presence of `MaybeUninit` which we use to implement `PointerGenerator`. Kani will detect the usage of `MaybeUninit` and it fail the verification.

Reply:
Can you perhaps include that clarification in the comment?